common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -4512,8 +4512,11 @@ static jstring NativeCrypto_get_X509_CRL_sig_alg_oid(JNIEnv* env, jclass, jlong
         return nullptr;
     }
 
-    int nid = X509_CRL_get_signature_nid(crl);
-    return ASN1_OBJECT_to_OID_string(env, OBJ_nid2obj(nid));
+    const X509_ALGOR *sig_alg;
+    X509_CRL_get0_signature(crl, nullptr, &sig_alg);
```

Comment:
(The output argument order between X509_get0_signature and X509_CRL_get0_signature is indeed flipped. No idea why. OpenSSL. Maybe I'll send them new APIs at some point because these are _atrocious_.)

Reply:
Fun....